### PR TITLE
add [Models\Service]::createService() #52

### PR DESCRIPTION
**Pull request by Kanonyy**

### CODE REVIEW

**Comment by Kanonyy** [comment]
Github -da tab=8 spaces olduğundan indentation səhv göstərir

**Comment by devalirzayev** [comment]
@Kanonyy gel Buffer, BusinessHour ve WeekDay-i de ayri modellere cixaraq.

example,
```php
$monday = new WeekDay();
$monday->setStartTime( "10:00" );
$monday->setEndTime( "09:00" );
$monday->save();

$tuesday = new WeekDay();
$tuesday->setStartTime( "10:00" );
$tuesday->setEndTime( "09:00" );
$tuesday->save();

// ... other days

$businessHours = new BusinessHour();
$businessHours->setMonday( $monday->getId() );
$businessHours->setTuesday( $tuesday->getId() );
// ... other days
$businessHours->setSunday( null );
$businessHours->save();

$buffer = new Buffer();
$buffer->setBeforeTime( 0 );
$buffer->setAfterTime( 60 );
$buffer->save();

$service = new Service();
$service->setName( "Test" );
// ... set other properties
$service->setBusinessHoursId( $businessHours->getId() );
$service->setBufferId( $buffer->getID() );
$service->save();
```

Hele ki, Controller terefini yazmiriq deye, bunlari createService methodunun icinde yaza bilerik (muveqqeti). Ve bir de example-da try/catchlari yazmamisham. Sualin olsa yaz, muzakire edek.

File: app/Models/Service.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        $stmt = self::getConnection()->prepare($sql);
+
+        foreach ($valuesNTypes as $k => $v) {
+            $stmt->bindValue(":$k", $v[0], $v[1]);
+        }
+
+		$stmt->execute();
+
+		return self::getConnection()->lastInsertId();
+    }
```
Comment (devalirzayev) [inline app/Models/Service.php:139]: bu method da service classinin icinde olmali deyil normalda, hansisa traitin icinde ola biler, Model classi da hemin traiti use eder. Bele olsa avtomatik butun modeller istifade ede biler onu. Ve ya DB classinin methodu ola biler, DB::exeSQL() formatinda.

**Review by devalirzayev — CHANGES_REQUESTED** [review]

**Review by devalirzayev — CHANGES_REQUESTED** [review]
.

File: app/Models/Service.php (hunk 1 above)
Comment (Kanonyy) [reply app/Models/Service.php:139]: DB class -na el vurmaq istemedim. PHP ise multiple inheritance desteklemir ve trait ne oldugunu indi oyrendim. PHP -ni on the fly oyrenirem evvel yazmamisam. Duzelderem

**Comment by Kanonyy** [comment]
> ```
> $monday = new WeekDay();
> $monday->setStartTime( "10:00" );
> $monday->setEndTime( "09:00" );
> $monday->save();
> 
> $tuesday = new WeekDay();
> $tuesday->setStartTime( "10:00" );
> $tuesday->setEndTime( "09:00" );
> $tuesday->save();
> 
> // ... other days
> 
> $businessHours = new BusinessHour();
> $businessHours->setMonday( $monday->getId() );
> $businessHours->setTuesday( $tuesday->getId() );
> // ... other days
> $businessHours->setSunday( null );
> $businessHours->save();
> 
> $buffer = new Buffer();
> $buffer->setBeforeTime( 0 );
> $buffer->setAfterTime( 60 );
> $buffer->save();
> 
> $service = new Service();
> $service->setName( "Test" );
> // ... set other properties
> $service->setBusinessHoursId( $businessHours->getId() );
> $service->setBufferId( $buffer->getID() );
> $service->save();
> ```

Men indi Buffer, BusinessHour ve WeekDay -i ozlerine uygun ayri modellere extract edim?

**Comment by devalirzayev** [comment]
> @devalirzayev
> 
> > ```
> > $monday = new WeekDay();
> > $monday->setStartTime( "10:00" );
> > $monday->setEndTime( "09:00" );
> > $monday->save();
> > 
> > $tuesday = new WeekDay();
> > $tuesday->setStartTime( "10:00" );
> > $tuesday->setEndTime( "09:00" );
> > $tuesday->save();
> > 
> > // ... other days
> > 
> > $businessHours = new BusinessHour();
> > $businessHours->setMonday( $monday->getId() );
> > $businessHours->setTuesday( $tuesday->getId() );
> > // ... other days
> > $businessHours->setSunday( null );
> > $businessHours->save();
> > 
> > $buffer = new Buffer();
> > $buffer->setBeforeTime( 0 );
> > $buffer->setAfterTime( 60 );
> > $buffer->save();
> > 
> > $service = new Service();
> > $service->setName( "Test" );
> > // ... set other properties
> > $service->setBusinessHoursId( $businessHours->getId() );
> > $service->setBufferId( $buffer->getID() );
> > $service->save();
> > ```
> 
> Men indi `createBuffer()`, `createBusinessHour()` ve `createWeekDay()` -i ozlerine uygun ayri modellere extract edim?

aha, artiq her model ozu cavabdeh olacaq database-e save etme prosesine

**Comment by Kanonyy** [comment]
> > @devalirzayev
> > > ```
> > > $monday = new WeekDay();
> > > $monday->setStartTime( "10:00" );
> > > $monday->setEndTime( "09:00" );
> > > $monday->save();
> > > 
> > > $tuesday = new WeekDay();
> > > $tuesday->setStartTime( "10:00" );
> > > $tuesday->setEndTime( "09:00" );
> > > $tuesday->save();
> > > 
> > > // ... other days
> > > 
> > > $businessHours = new BusinessHour();
> > > $businessHours->setMonday( $monday->getId() );
> > > $businessHours->setTuesday( $tuesday->getId() );
> > > // ... other days
> > > $businessHours->setSunday( null );
> > > $businessHours->save();
> > > 
> > > $buffer = new Buffer();
> > > $buffer->setBeforeTime( 0 );
> > > $buffer->setAfterTime( 60 );
> > > $buffer->save();
> > > 
> > > $service = new Service();
> > > $service->setName( "Test" );
> > > // ... set other properties
> > > $service->setBusinessHoursId( $businessHours->getId() );
> > > $service->setBufferId( $buffer->getID() );
> > > $service->save();
> > > ```
> > 
> > 
> > Men indi `createBuffer()`, `createBusinessHour()` ve `createWeekDay()` -i ozlerine uygun ayri modellere extract edim?
> 
> aha, artiq her model ozu cavabdeh olacaq database-e save etme prosesine

@devalirzayev 
`create*()` ile duzeldim ya `get*()`/`set*()`/`save*()`? Eger `get*()`/`set*()`/`save*()` onda return value `save` de `id` olacaq digerleri ise hec ne return etmeyecek?

**Comment by devalirzayev** [comment]
digərləri deyəndə nəyi nəzərdə tutursan? save() void olacaq successdə

**Comment by devalirzayev** [comment]
service-in id-ni götürmək istəsək, onda da $service->getId() edəcəyik

**Comment by Kanonyy** [comment]
yeni commit atdim amma save -den sonra deyisiklik etsek yeni row yaradir.

**Comment by Kanonyy** [comment]
save -i duzeldecem amma gor basqa her sey qane edir?

**Comment by devalirzayev** [comment]
Okay gorunur, lazim olsa diger hisseleri daha sonra deyisherik, bu task ucun kifayet edir bunlar 👍🏻

**Review by devalirzayev — APPROVED** [review]

**Pull request merged**